airbyte_cdk/sources/declarative/decoders/json_decoder.py

```diff
@@ -70,8 +77,17 @@ class JsonlDecoder(Decoder):
     def is_stream_response(self) -> bool:
         return True
 
-    def decode(self, response: requests.Response) -> Generator[Mapping[str, Any], None, None]:
+    def decode(self, response: requests.Response) -> Generator[MutableMapping[str, Any], None, None]:
         # TODO???: set delimiter? usually it is `\n` but maybe it would be useful to set optional?
         #  https://github.com/airbytehq/airbyte-internal-issues/issues/8436
         for record in response.iter_lines():
             yield orjson.loads(record)
+
+
+@dataclass
+class GzipJsonDecoder(JsonDecoder):
```

Comment:
Should we make this more generic? It feels like we should have a GzipDecoder that takes another decoder as once decompressed, the format could be anything. The issue right now is that the interface relying on `request.Response`. Hence, it feels like there is an abstraction or a concept missing...

Reply:
> Should we make this more generic?

I'm not sure about it. All implementations are inherited from `Decoder`, which has only 1 method `decode`

>[!NOTE]   
 Decoder strategy to transform a requests.Response into a Mapping[str, Any]

We _could_ make the decoder truly generic by wrapping the response content in a byte stream (e.g., `BytesIO`). However, this would require implementing custom iterators for `lines` or `content`, significantly increasing code complexity.

Reply:
The downside is that we will have to create all possible combinations of decompression and format. I'm fine for leaving this as is for now but it doesn't seem like something that will scale though